### PR TITLE
Simplify model parameters

### PR DESCRIPTION
**Pull request by streeve**
Reorder params to match between creation functions and base model. Pass in order of:

- Model specific parameters
- Required "main" base parameters (CG is a slightly different use case)
- Optional base parameters

Repeating values in the creation factory function is still annoying, but this at least removes repetition from the derived models

### CODE REVIEW

File: hydragnn/models/create.py [hunk 1]
```diff
@@ -70,69 +71,72 @@ def create_model(
...
         model = GATStack(
-            input_dim=input_dim,
-            output_dim=output_dim,
-            hidden_dim=hidden_dim,
-            num_nodes=num_nodes,
-            num_conv_layers=num_conv_layers,
-            output_type=output_type,
-            config_heads=output_heads,
+            6,
+            0.05,
```
Comment (pzhanggit) [inline hydragnn/models/create.py:105]: A little confusing to see two number here. Maybe defining ` heads = 6,
        negative_slope = 0.05` here and then pass it to GATStack or set the variables inside GATStack?

**Review by pzhanggit — COMMENTED** [review]
Why no test running for this PR?

**Comment by streeve** [comment]
@pzhanggit final review?

**Comment by streeve** [comment]
lol nevermind

**Review by pzhanggit — APPROVED** [review]

**Pull request merged**